### ISSUE TITLE
taskcat test run --keep-failed -> ValueError status is not a valid property of <class 'taskcat._cfn.stack.Stack'>

### ISSUE THREAD

**Issue by tekdj7** [aws-quickstart/taskcat#660]
**Describe the bug**
Ran `taskcat test run --keep-failed` across (21) AWS Commercial Regions.  It did not delete ANY of the CloudFormation Stacks (both successful & failed).

- After `reporting on` messages come out, see below messages:
  - `[INFO   ] : deleting successful stacks`
  - `[ERROR  ] : ValueError status is not a valid property of <class 'taskcat._cfn.stack.Stack'>`

**To Reproduce**
1. Are you testing a *QuickStart* or *Custom template*? `No`
2. Attach or link a copy of the template if possible (**remove any sensitive info**) 
3. Provide the parameters that you passed. (**remove any sensitive info**)
4. How did you install taskcat? (docker or pip3) `pipx`
5. Are you using a *profile*, *an instance role* or  *access keys* to run taskcat? `profile`
6. Is your AWS environment configured via `aws configure`? `yes`

**Expected behavior**
- I was expecting that it would delete the stacks in all AWS regions where stack deployed successfully, and that it would not delete the stacks in the AWS regions where stack deployment failed.
- FYI, it works properly when I do `taskcat test run`, and then stacks are deleted in all AWS regions (both successful & failed stacks)

**Version (Please make sure you are running the latest version of taskcat)
 - Taskcat Version: `0.9.23`
 - Python3 Version: `3.8.7`

**Comment by shadycuz** [comment]
This is something I can help with. Let me do some digging around and I will let you know when I have submitted a fix.

**Comment by andrew-glenn** [comment]
❤️

**Comment by shadycuz** [comment]
@andrew-glenn Trying to track this down but its a harder than I expected. 

TLDR: I found the fix you can skip to the bottom if you want

It starts in [_cfn_test.py](https://github.com/aws-quickstart/taskcat/blob/2072dd5182afe7a260cb7d6e2af132ee09490fb5/taskcat/testing/_cfn_test.py#L148) where we try to delete only the completed stacks. We do that by filtering out stacks using this dictionary `{"status": "CREATE_COMPLETE"}`. Where the key will be the attribute of the stack as in `stack.status` and the value is the condition we are looking for. 

 So the error is actually coming from the [filter function](https://github.com/aws-quickstart/taskcat/blob/2072dd5182afe7a260cb7d6e2af132ee09490fb5/taskcat/_cfn/stack.py#L160). But what I can't figure out is how the stack object doesn't have the `status` attribute as its [clearly defined](https://github.com/aws-quickstart/taskcat/blob/2072dd5182afe7a260cb7d6e2af132ee09490fb5/taskcat/_cfn/stack.py#L256). 
 
 I can only think of two causes. Either a object that wasn't a stack got added to stackers self.stacks or somehow a stack doesn't have the `status` attribute but neither of these make sense to me?

Okay so I think this is by [design](https://bugs.python.org/issue2040). Apparently a property behaves as a Class attribute and not a instance attribute! 

You can see it with this little test. 

```
>>> class foo():
...     def __init__(self):
...             self._status = 'test'
...     @property
...     def status(self):
...             return self._status
... 
>>> test = foo()
>>> print(test.__dict__)
{'_status': 'test'}
>>> print(foo.__dict__)
{'__module__': '__main__', '__init__': <function foo.__init__ at 0x7fdf8d479598>, 'status': <property object at 0x7fdf8d46cbd8>, '__dict__': <attribute '__dict__' of 'foo' objects>, '__weakref__': <attribute '__weakref__' of 'foo' objects>, '__doc__': None}
````

I know I have used keep failed and it worked but I don't remember when? 


## TLDR
Okay I see the problem now. I'm really glad I wrote all this out or I wouldn't have found it lol. I don't know who this [guy is](https://github.com/aws-quickstart/taskcat/commit/c16ea216337d6e7e2274fea9013e2559824ead7d) but he broke it ;). 

I will submit a fix and hopefully a working test case.
